### PR TITLE
Direct3d11 small fix and improvement

### PR DESCRIPTION
**Pull request by mrvux**
Hello, just a small change in Direct3D11.

* Fix GetUnorderedAccess view call in case no depth buffer is bound
* Add ComArray binding for CSSetUnorderedAccessViews (same as Constant buffer/SRV counterpart)

Thanks

### CODE REVIEW

File: Source/SharpDX.Direct3D11/DeviceContext.ComputeShaderStage.cs [hunk 1]
```diff
@@ -105,5 +105,16 @@ public unsafe void SetUnorderedAccessViews(int startSlot, SharpDX.Direct3D11.Uno
...
+
+        /// <summary>	
+        /// Sets an array of views for an unordered resource.	
+        /// </summary>	
+        /// <param name="startSlot">Index of the first element in the zero-based array to begin setting. </param>
+        /// <param name="unorderedAccessViews">A reference to an array of <see cref="SharpDX.Direct3D11.UnorderedAccessView"/> references to be set by the method. </param>
+        public unsafe void SetUnorderedAccessViews(int startSlot, ComArray<SharpDX.Direct3D11.UnorderedAccessView> unorderedAccessViews, int[] uavInitialCounts)
+        {
+            fixed (void* puav = uavInitialCounts)
+                SetUnorderedAccessViews(startSlot, unorderedAccessViews == null ? 0 : unorderedAccessViews.Length, unorderedAccessViews.NativePointer, (IntPtr)puav);
```
Comment (h1cks) [inline Source/SharpDX.Direct3D11/DeviceContext.ComputeShaderStage.cs]: Won't you get a null pointer exception on unorderedAccessViews.NativePointer for SetUnorderedAccessViews as you are not testing that it is also Null?  You're testing length for a null, but you're not testing NativePointer for the same possible case.

Comment (mrvux) [reply Source/SharpDX.Direct3D11/DeviceContext.ComputeShaderStage.cs]: Indeed, thinking of that in pretty much all existing DeviceContext "Set[]" which use ComArray, there is no null check at all, so could actually remove the first check.

Comment (h1cks) [reply Source/SharpDX.Direct3D11/DeviceContext.ComputeShaderStage.cs]: There are a couple of such cases, agreed, but I think if you did check Null Pointer on unorderedAccessViews then that woudl be fine.  OR for removing ambiguity, set the code as:

` fixed (void* puav = uavInitialCounts)
			{
				if (unorderedAccessViews != null)
				{
					SetUnorderedAccessViews(startSlot, unorderedAccessViews.Length, uorderedAccessViews.NativePointer, (IntPtr)puav);
				}
				else
				{
					SetUnorderedAccessViews(startSlot, 0, null, (IntPtr)puav);
				}
			}`

I think the null pointer checks are correct that you are doing.  I think some expansion to protect against a legitimate use case where the UAV is set to null.  But yes, there are some checks missed in the code.
